src/components/chat.d.ts

```diff
@@ -0,0 +1,22 @@
+import { RootEmitter } from '../modules/root_emitter';
+
+export declare class Chat extends RootEmitter {
+  constructor(chatEngine: any, channel: any, isPrivate: boolean, autoConnect: boolean, meta: any, group: string);
+
+  onHereNow(status: any, response: any);
+  objectify(): any;
+  invite(user: any): void;
+  onPresence(presenceEvent: any): void;
+  update(data: any): void;
+  emit(event: string, data: any): void;
+  createUser(uuid: string, state: any): void;
+  userUpdate(uuid: string, state: any): void;
+  leave(): void;
+  userLeave(uuid: string): void;
+  userDisconnect(uuid: string): void;
+  setState(state: any): void;
+  search(config: any): void;
+  onConnectionReady(): void;
+  getUserUpdates(): void;
+  connect(): void;
```

Comment:
Gosh, we really have to maintain this list?

Reply:
Did you use something like https://github.com/Microsoft/dts-gen?

Reply:
that would be ideal but it’s only necessary the main

Reply:
Can we use jsdoc to generate the output?
https://github.com/englercj/tsd-jsdoc

Reply:
I have another pull request that changes methods in some of these classes. Maintaining the maps by hand will be tedious and we don't have a good way of testing it.

Reply:
yes, in this point you’re right but we can only use the d.ts for the index file, in this way it’s not necessary to make changes, what do you think about this?